### PR TITLE
[RUMF-845] remove slimDOM option and enable strategies to ignore elements

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

A bit less code (removed option) and lighter snapshots ("slim DOM" option always enabled)

## Changes

* Remove the "slim DOM" option
* Always apply all "slim DOM" strategies
* Add some unit testings

## Testing

Unit, manual

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/master/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.spec.ts [hunk 1]
```diff
@@ -73,3 +73,52 @@ describe('absolute url to stylesheet', () => {
...
+
+describe('serializeNodeWithId', () => {
+  describe('ignores some nodes', () => {
+    const defaultOptions = {
+      doc: document,
+      skipChild: false,
+      map: {},
+    }
+
+    it('does not same ignored nodes in the map', () => {
```
Comment (webNeat) [inline packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.spec.ts:85]: I think there is a typo here, should it be `does not save ignored nodes in the map` ?

Comment (BenoitZugmeyer) [reply packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.spec.ts:85]: Good catch!

**Review by webNeat — APPROVED** [review]
Nice!

File: packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.spec.ts [hunk 2]
```diff
@@ -73,3 +73,52 @@ describe('absolute url to stylesheet', () => {
...
+      map: {},
+    }
+
+    it('does not save ignored nodes in the map', () => {
+      const map = {}
+      serializeNodeWithId(document.createElement('script'), { ...defaultOptions, map })
+      expect(map).toEqual({})
+    })
+
+    it('sets the serialized node id to IGNORED_NODE', () => {
```
Comment (bcaudan) [inline packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.spec.ts:91]:
```suggestion
    it('sets ignored serialized node id to IGNORED_NODE', () => {
```

File: packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.spec.ts [hunk 3]
```diff
@@ -73,3 +73,52 @@ describe('absolute url to stylesheet', () => {
     expect(absoluteToStylesheet(`url('')`, href)).toEqual(`url('')`)
   })
 })
+
+describe('serializeNodeWithId', () => {
```
Comment (bcaudan) [inline packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.spec.ts:77]: cf BS unit tests, adding those tests expose some IE incompatibility in snapshot code (array.from)

**Review by bcaudan — APPROVED** [review]

**Pull request merged**